### PR TITLE
Feature - Add Elemental Blocks

### PR DESCRIPTION
**Pull request by ssmarco**

### CODE REVIEW

File: src/Controller/FabricatorAPIController.php [hunk 1]
```diff
@@ -132,42 +142,39 @@ public function getBlock(HTTPRequest $request)
...
-            'BackgroundColour' => 'aubergine',
-            'CTASummary' => 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
-        ];
-
-        $generatedTemplate = SSViewer::execute_template(
-            $className,
-            $this->owner,
-            $templateData
-        );
+        // $instance->publishSingle(); // default is Save Only on Staging
```
[the remark this answers is not in the document]
Comment (ssmarco) [reply src/Controller/FabricatorAPIController.php]: As discussed, we need to Save at least similar on how CMS works when creating a new record.

File: client/stores/store.js [hunk 2]
```diff
@@ -286,22 +303,27 @@ export const useStore = defineStore('store', {
...
     /**
-     * Add two way 'bindings' to allow the CMS auther to click the elemental block and open it in the Fabricator vue
+     * Add two-way 'bindings' to allow the CMS author to click the elemental block and open it in the Fabricator vue
      */
     mapEvents() {
       this.page.blocks.map(block => {
         let anchor = block.Anchor;
         let type = block.Type;
+
+        if (block.HasEvents) return;
```
Comment (dannidickson) [inline client/stores/store.js:313]: Nice touch!

File: src/Controller/FabricatorAPIController.php [hunk 3]
```diff
@@ -132,44 +143,51 @@ public function getBlock(HTTPRequest $request)
...
+        $allFields = DataObject::getSchema()->fieldSpecs($instance);
+
+        foreach ($allFields as $field => $fieldSpec) {
+            $fieldObj = $instance->dbObject($field);
+
+            if ($fieldObj instanceof DBHTMLText) {
+                $instance->{$field} = '<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, </p>'
+                    . '<p>sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>';
+            }
+        }
```
Comment (dannidickson) [inline src/Controller/FabricatorAPIController.php:167]: As discussed this looks great for an MVP. I do think we should use the `getCMSFields`, or the fabricator's cleaned up version `getFrontEndFields()`

File: client/views/BlockAdd.vue [hunk 4]
```diff
@@ -40,26 +40,33 @@ const store = useStore();
...
     store.newBlock(selectedBlock)
-    .then(template => {
-        console.log(blockAnchor);
-        Clippy.addBlockToDOM(blockAnchor, template);
-    });
+      .then(template => {
+          if (template) {
+              Clippy.addBlockToDOM(blockAnchor, template);
+              store.mapEvents();
+              store.closeBlock();
```
Comment (dannidickson) [inline client/views/BlockAdd.vue:57]: Maybe instead of closing, we open the block?

**Review by dannidickson — APPROVED** [review]
Im happy to approve it as is for MVP, and we can refine it as we go. Obviously because its a complex UI program it will evolve.

File: src/Controller/FabricatorAPIController.php (hunk 3 above)
Comment (ssmarco) [reply src/Controller/FabricatorAPIController.php]: :+1: I forgot you mentioned about Fabricator's similar method.

File: client/views/BlockAdd.vue (hunk 4 above)
Comment (ssmarco) [reply client/views/BlockAdd.vue:57]: That looks like a good idea. Had a look how it is implemented on `BlockList.vue` but it creates another round-trip to the server and then recreate the blocks again. I think we need to revisit the current approach.

Comment (ssmarco) [reply client/views/BlockAdd.vue:57]: For the meantime I will implement the round-trip which is enough for MVP

**Pull request merged**